### PR TITLE
cherrypick e2e changes regarding maverick from tm master

### LINKED ISSUE #292: e2e: cherry-pick latest changes from tendermint master
We were planning to hold off merging in changes from tendermint master until 0.35 lands but @Wondertan brought up a good point: we are currently doing changes in the Consensus package and we would have repeat thos in [maverick](https://github.com/lazyledger/lazyledger-core/tree/bcdbc1b5e834d7ba58b0faf1673ff249eba6f4d1/test/maverick) whose binary in the p2p package was recently removed in tendermint: https://github.com/tendermint/tendermint/pull/6099. See also this issue: https://github.com/tendermint/tendermint/issues/5575.

I think it would make sense to look into cherry-picking the corresponding changes:
 - https://github.com/tendermint/tendermint/pull/6099
 - https://github.com/tendermint/tendermint/pull/6276 

That sounds easier as it is (or might be) because there might be more changes that will cause merge conficts (both from the LazyLedger as well as the tendermint side).

### PR DESCRIPTION
**Pull request by liamsi**
## Description

This PR only takes on the easy part of #292 (removing maverick) but I actually think we do not need to get https://github.com/tendermint/tendermint/pull/6276 and related changes to close #292 and unblock work on the consensus reactor (#293).

Each commit here was cherry-picked from tendermint master (you can look at corresponding squash merged PRs in tendermint). I've included some that seemed to further help with e2e flakiness. So hopefully this also improves e2e test reliability for this repo.

The last commit (6628237) bumps the codecov action version (see https://twitter.com/thomasrockhu/status/1384619989634797576).

Closes: #292

### CODE REVIEW

**Review by evan-forbes — APPROVED** [review]
see you soon, maverick

**Comment by Wondertan** [comment]
> see you soon
Are we going to bring it back?

File: .github/workflows/coverage.yml [hunk 1]
```diff
@@ -121,7 +121,7 @@ jobs:
       - run: |
           cat ./*profile.out | grep -v "mode: atomic" >> coverage.txt
         if: env.GIT_DIFF
-      - uses: codecov/codecov-action@v1.4.0
+      - uses: codecov/codecov-action@v1.4.1
```
Comment (Wondertan) [inline .github/workflows/coverage.yml:124]: Is this related to the change?

File: test/e2e/app/main.go [hunk 2]
```diff
@@ -62,11 +59,12 @@ func run(configFile string) error {
 	// Start app server.
 	switch cfg.Protocol {
 	case "builtin":
-		if len(cfg.Misbehaviors) == 0 {
-			err = startNode(cfg)
-		} else {
-			err = startMaverick(cfg)
-		}
+		// FIXME: Temporarily remove maverick until it is redesigned
```
Comment (Wondertan) [inline test/e2e/app/main.go:62]: Ok, I see

> Maverick: I'll be back

**Review by Wondertan — APPROVED** [review]

File: test/e2e/app/main.go (hunk 2 above)
Comment (Wondertan) [reply test/e2e/app/main.go:62]: Does the tm team plan to redesign maverick or we? IMO: good redesign would require consensus redesign in a more modular way. Copying code is nah

File: .github/workflows/coverage.yml (hunk 1 above)
Comment (liamsi) [reply .github/workflows/coverage.yml:124]: No, but otherwise:
image

File: test/e2e/app/main.go (hunk 2 above)
Comment (liamsi) [reply test/e2e/app/main.go:62]: Yeah, it redesigned it already as far as I undertand. It's now part of the other e2e tests (at least the most relevant parts).
